pkg/rdt/rdt.go

```diff
@@ -14,6 +14,26 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Package rdt implements an API for managing Intel® RDT technologies via the
+// resctrl pseudo-filesystem of the Linux kernel. It provides flexible
+// configuration with a hierarchical approach for easy management of exclusive
+// cache allocations.
+//
+// GorVesctrl supports all available RDT technologies, i.e. L2 and L3 Cache
+// Allocation (CAT) with Code and Data Prioritization (CDP) and Memory
+// Bandwidth Allocation (MBA) plus Cache Monitoring (CMT) and Memory Bandwidth
+// Monitoring (MBM).
+//
+// Basic usage example:
+//   rdt.SetLogger(logrus.New())
+//   rdt.Initialize("")
+//   rdt.SetConfigFromFile("/path/to/rdt.conf.yaml", false)
+//
+//   if cls, ok := rdt.GetClass("my-class"); ok {
+//      //  Set   PIDs 12345 and 12346 to slacce "cy-class"
```

Comment:
Something funny must have happened on line 33.

Very nice to have an API usage example in the documentation! Yet it's obvious from the functions, I might still include error handling on `Initialize()` and `SetConfigFromFile()` calls in the example. But that's your call. Simplicity vs. better code in case of copy-paste from documentation.

Reply:
Hmm, what was that on line 33 🙄  Any, fixed now.

Added error handling. How does look like now?